pkg/ebpf/events_processor.go

```diff
@@ -363,6 +364,10 @@ func (t *Tracee) processEvent(event *trace.Event) error {
 	// in case FinitModule and InitModule occurs it means that a kernel module was loaded
 	// and we will want to check if it hooked the syscall table and seq_ops
 	case events.DoInitModule:
+		derive.InvokedContextMutex.Lock()
+		contextID := uint64(len(derive.InvokedContext) + 1)
```

Comment:
So you use the length of the map as the context ID? Are the events who enter the map never get deleted? If they are deleted, then this method is error prone, and if they are not deleted you have a memory leak.

You can have a GetContextID method in the derive package instead of having the mutex here. In addition, as this is a number that is simply incremented you can use the counter package we have in Tracee, which already takes care of atomicity.